Map a whole bunch of stuff

pyrox0 opened the pull request:
As the title says, n/m/command is now fully mapped to the best of my ability, as well as n/m/block to 99%!(8433/8517)

Most of n/m/block was adding CODEC entries for all the blocks, which took a while but shouldn't need to be done again in the future.

n/m/command wasn't too bad, just a bunch of thorny logic to reason through, but I got there in the end.

n/m/scoreboard is essentially fully mapped as well, with only one mysterious field in the Scoreboard class missing. Required pulling a bunch of unmapped stuff out, but got that untangled as well, so that should be done too!

n/m/client/realms also has a bunch of thorny logic, as is to be expected, but pulled that out and it's good now

n/m/resource was trivial.

Feedback is of course welcome. I know there are a few conflicts(or at least similar/identical changes) in #523 , but I'm glad to remove my changes that conflict with the ones in that PR if needed!

pyrox0 committed:
map n/m/block as much as I can

Now at 99.0%! Up about 4.5% from when I started :)

pyrox0 committed:
fully map n/m/command

100%! :tada:

pyrox0 committed:
fully map n/m/scoreboard

99.8%(471/472), only one that's missing is a mysterious constant in Scoreboard that Yarn doesn't have a name for either, so I didn't map it.

pyrox0 committed:
fully map n/m/client/realms

kaBAM, another mapping bites the dust

pyrox0 committed:
fully map n/m/resource

*DJ Khaled voice* Another one

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
 	METHOD m_agroqyjs stainedGlassOf (Lnet/minecraft/unmapped/C_arllgqae;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 color
+	METHOD m_cejxlkjq pillarOf (Lnet/minecraft/unmapped/C_mlrprqox;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 color
```
I preffer #523's `netherStemOf` for this.
Mojang calls this `netherStem`, I think this falls under [CONVENTIONS.md#mojang-names](https://github.com/QuiltMC/quilt-mappings/blob/1.20.2-pre4/CONVENTIONS.md#mojang-names), where the example given is don't rename `BedrockBlock` `NoSpawningBlock`.

I prefer the parameter `color` in this PR though.

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
+	METHOD m_cejxlkjq pillarOf (Lnet/minecraft/unmapped/C_mlrprqox;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 color
 	METHOD m_dsrskptj shulkerBoxOf (Lnet/minecraft/unmapped/C_arllgqae;Lnet/minecraft/unmapped/C_mlrprqox;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 dyeColor
```
I prefer this to `color` in #523

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
+		ARG 3 entity
 	METHOD m_fwewvidj stairOf (Lnet/minecraft/unmapped/C_mmxmpdoq;)Lnet/minecraft/unmapped/C_mmxmpdoq;
 	METHOD m_ghajpyfj register (Lnet/minecraft/unmapped/C_xhhleach;Lnet/minecraft/unmapped/C_mmxmpdoq;)Lnet/minecraft/unmapped/C_mmxmpdoq;
 	METHOD m_gtmmdkuy leaveOf (Lnet/minecraft/unmapped/C_aevintex;)Lnet/minecraft/unmapped/C_mmxmpdoq;
```
```suggestion
	METHOD m_gtmmdkuy leavesOf (Lnet/minecraft/unmapped/C_aevintex;)Lnet/minecraft/unmapped/C_mmxmpdoq;
```

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
 	METHOD m_fwewvidj stairOf (Lnet/minecraft/unmapped/C_mmxmpdoq;)Lnet/minecraft/unmapped/C_mmxmpdoq;
 	METHOD m_ghajpyfj register (Lnet/minecraft/unmapped/C_xhhleach;Lnet/minecraft/unmapped/C_mmxmpdoq;)Lnet/minecraft/unmapped/C_mmxmpdoq;
 	METHOD m_gtmmdkuy leaveOf (Lnet/minecraft/unmapped/C_aevintex;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 sounds
```
I prefer #523's `soundGroup`

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
 	METHOD m_hjyhmutz buttonOf (Lnet/minecraft/unmapped/C_hgpogkhy;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 type
 	METHOD m_ilocxcob candleOf (Lnet/minecraft/unmapped/C_mlrprqox;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 color
```
I prefer this to `mapColor` in #523

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
 	METHOD m_ilocxcob candleOf (Lnet/minecraft/unmapped/C_mlrprqox;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 color
+	METHOD m_kcksfacw (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_ogavsvbr;)Z
+		ARG 3 entity
```
```suggestion
		ARG 3 entityType
```
I think it's better to say the whole thing rather than `entity` here or `type` in #523.
It also matches the params that come later in this file.

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
 	METHOD m_sfzscxyj createPiston (Z)Lnet/minecraft/unmapped/C_mmxmpdoq;
 		ARG 0 sticky
 	METHOD m_sidkwvli createStoneButton ()Lnet/minecraft/unmapped/C_mmxmpdoq;
+	METHOD m_tetaodtf initShapeCaches ()V
```
I can't check the source to confirm rn, but #523 uses `rebuildShapeCache` which is closer to Mojang's `rebuildCache`.

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
 		ARG 0 sticky
 	METHOD m_sidkwvli createStoneButton ()Lnet/minecraft/unmapped/C_mmxmpdoq;
+	METHOD m_tetaodtf initShapeCaches ()V
+	METHOD m_uvuszloo pillarOf (Lnet/minecraft/unmapped/C_mlrprqox;Lnet/minecraft/unmapped/C_mlrprqox;Lnet/minecraft/unmapped/C_aevintex;)Lnet/minecraft/unmapped/C_mmxmpdoq;
```
I prefer #523's `logOf` for the same reason I gave for `m_cejxlkjq pillarOf`.
For the params, #523 uses top/side, which I can't assess without source, but I prefer just `Color` in this PR as opposed to #523's `MapColor`.
I prefere #523's`soundGroup` to `sounds`, though.

supersaiyansubtlety commented on mappings/net/minecraft/block/Blocks.mapping:
```diff
@@ -1,21 +1,45 @@
 	METHOD m_wgtmwkrm bedOf (Lnet/minecraft/unmapped/C_arllgqae;)Lnet/minecraft/unmapped/C_mmxmpdoq;
+		ARG 0 color
 	METHOD m_wjfyifdb nonSolid (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_peaveboq;Lnet/minecraft/unmapped/C_hynzadkk;)Z
+	METHOD m_zfebkwem pillarOf (Lnet/minecraft/unmapped/C_mlrprqox;Lnet/minecraft/unmapped/C_mlrprqox;)Lnet/minecraft/unmapped/C_mmxmpdoq;
```
Same as `m_uvuszloo pillarOf` (and `m_cejxlkjq pillarOf`).

supersaiyansubtlety commented on mappings/net/minecraft/block/Crafter.mapping:
```diff
@@ -1 +1,13 @@
 CLASS net/minecraft/unmapped/C_dmnruvyr net/minecraft/block/Crafter
+	FIELD f_agoiltbq TRIGGER_DELAY I
```
I prefer this to #523's `CRAFTING_TICK_DELAY`

supersaiyansubtlety replied:
No I prefer your original `TRIGGER_DELAY`!

pyrox0 replied:
Alright, will update this later!

Edit: Updated!

supersaiyansubtlety commented on mappings/net/minecraft/block/Degradable.mapping:
```diff
@@ -1,5 +1,7 @@
 CLASS net/minecraft/unmapped/C_pfqhyrqm net/minecraft/block/Degradable
 	FIELD f_fzcgonwn SCAN_DISTANCE I
+	METHOD m_hgduqvhl tickDegradable (Lnet/minecraft/unmapped/C_txtbiemp;Lnet/minecraft/unmapped/C_bdwnwhiu;Lnet/minecraft/unmapped/C_hynzadkk;Lnet/minecraft/unmapped/C_rlomrsco;)V
```
I prefer #523's `tickDegradation`

supersaiyansubtlety commented on mappings/net/minecraft/block/Fertilizable.mapping:
```diff
@@ -1,4 +1,5 @@
 CLASS net/minecraft/unmapped/C_svvihwsq net/minecraft/block/Fertilizable
+	METHOD m_lprnjghr getFertilizeParticlePos (Lnet/minecraft/unmapped/C_hynzadkk;)Lnet/minecraft/unmapped/C_hynzadkk;
```
Mojang and #523 both call this `getParticlePos` which I prefer

supersaiyansubtlety commented on mappings/net/minecraft/block/PinkPetalsBlock.mapping:
```diff
@@ -2,4 +2,6 @@ CLASS net/minecraft/unmapped/C_onsqqphm net/minecraft/block/PinkPetalsBlock
 	FIELD f_crwyarkl MAX_PETAL_AMOUNT I
 	FIELD f_hvhlkcdp FACING Lnet/minecraft/unmapped/C_tcujuvjb;
 	FIELD f_ierdplen AMOUNT Lnet/minecraft/unmapped/C_vltzvhxi;
+	FIELD f_jprjgbfu FACING_AND_AMOUNT_TO_SHAPE Ljava/util/function/BiFunction;
```
```suggestion
	FIELD f_jprjgbfu SHAPE_BY_PROPERTIES
 Ljava/util/function/BiFunction;
```
This is what Mojang calls it, and it follows the `valuesByKeys` format from [CONVENTIONS.md#collections](https://github.com/QuiltMC/quilt-mappings/blob/1.20.2-pre4/CONVENTIONS.md#collections). I also prefer the conciseness of `PROPERTIES` to `FACING_AND_AMOUNT`

supersaiyansubtlety commented on mappings/net/minecraft/block/WallPiglinSkullBlock.mapping:
I think this class should be `PiglinWallHeadBlock` based on the translation key `piglin_wall_head`

supersaiyansubtlety commented on mappings/net/minecraft/block/WallPlayerSkullBlock.mapping:
I think this class should be `PlayerWallHeadBlock` based on the translation key `player_wall_head`

supersaiyansubtlety commented on mappings/net/minecraft/block/WallRedstoneTorchBlock.mapping:
I think this class should be `RedstoneWallTorchBlock` based on the translation key `redstone_wall_torch`

supersaiyansubtlety commented on mappings/net/minecraft/block/WallWitherSkullBlock.mapping:
I think this class should be `WitherSkeletonWallSkullBlock` based on the translation key  `wither_skeleton_wall_skull`

supersaiyansubtlety commented on mappings/net/minecraft/block/entity/JigsawBlockEntity.mapping:
```diff
@@ -13,19 +17,25 @@ CLASS net/minecraft/unmapped/C_epwegwuj net/minecraft/block/entity/JigsawBlockEn
 	METHOD m_sushgpja setTargetPool (Lnet/minecraft/unmapped/C_ncpywfca;)V
 		ARG 1 target
+	METHOD m_taysyesg setPlacementPriority (I)V
+		ARG 1 placementPriority
 	METHOD m_ummebuzz getPool ()Lnet/minecraft/unmapped/C_xhhleach;
 	METHOD m_uvspztdw setAttachmentType (Lnet/minecraft/unmapped/C_ncpywfca;)V
 		ARG 1 value
 	METHOD m_vallwvdc getFinalState ()Ljava/lang/String;
+	METHOD m_vmdfkedj setSelectionPriority (I)V
+		ARG 1 selectionPriority
```
```suggestion
		ARG 1 priority
	METHOD m_ummebuzz getPool ()Lnet/minecraft/unmapped/C_xhhleach;
	METHOD m_uvspztdw setAttachmentType (Lnet/minecraft/unmapped/C_ncpywfca;)V
		ARG 1 value
	METHOD m_vallwvdc getFinalState ()Ljava/lang/String;
	METHOD m_vmdfkedj setSelectionPriority (I)V
		ARG 1 priority
```
I think the method names make the param names a little redundant.

supersaiyansubtlety commented on mappings/net/minecraft/block/sign/SignType.mapping:
```diff
@@ -5,6 +5,7 @@ CLASS net/minecraft/unmapped/C_xlaykyai net/minecraft/block/sign/SignType
 	FIELD f_khfupynz DARK_OAK Lnet/minecraft/unmapped/C_xlaykyai;
 	FIELD f_lxkmsfqj MANGROVE Lnet/minecraft/unmapped/C_xlaykyai;
 	FIELD f_nldrovwk WARPED Lnet/minecraft/unmapped/C_xlaykyai;
+	FIELD f_npkrgudo SIGN_TYPE_CODEC Ljava/util/Map;
```
```suggestion
	FIELD f_npkrgudo CODEC Ljava/util/Map;
```

pyrox0 replied:
There's already an autogenerated `CODEC` field right below this, and it's not a Codec anyways, so changed it to be called `TYPES`, since it's a map of all the sign types.

supersaiyansubtlety reviewed (changes requested):
I tried to compare and comment on any conflicts between this PR and #523's block changes.

I also requested some changes to some wall block class names (and a few other misc blocks) that I only noticed because this PR maps their codec fields. I didn't use github's suggestion feature for class names since they need to be changed in enigma.

pyrox0 commented:
Updated the blocks commit based on the suggested changes!

Edit: I'm aware of the lint errors, working on fixing them, will hopefully have a better set of mappings uploaded by tomorrow.

Edit 2: Fixed the lint errors!

IotaBread committed:
Merge branch '23w51b' into updates

IotaBread committed:
Fix indentation in Crafter.mapping
